AndroidSDKTests/src/test/java/com/leanplum/LeanplumInboxMessageTest.java

```diff
@@ -0,0 +1,109 @@
+package com.leanplum;
+
+import com.leanplum.__setup.AbstractTest;
+import com.leanplum._whitebox.utilities.ResponseHelper;
+import com.leanplum.internal.Constants;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests covering Inbox Messages.
+ * @author Sayaan Saha
+ */
+
+public class LeanplumInboxMessageTest extends AbstractTest {
+  LeanplumInbox leanplumInbox;
+  @Before
+  public void setUp() {
+    leanplumInbox = Leanplum.getInbox();
+    setupSDK(mContext, "/responses/simple_start_response.json");
+  }
+
+  @Test
+  public void testMessageCreate() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, true);
+
+    LeanplumInboxMessage message = LeanplumInboxMessage.createFromJsonMap("message##Id", map);
+    assertEquals("message##Id", message.getMessageId());
+    assertEquals(delivery, message.getDeliveryTimestamp());
+    assertEquals(expiration, message.getExpirationTimestamp());
+    assertTrue(message.isRead());
+    assertNull(message.getData());
+
+    assertNull(message.getImageFilePath());
+    assertNull(message.getImageUrl());
+  }
+
+  @Test
+  public void testReadAndUpdateMessageCount() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, false);
+    LeanplumInboxMessage message = LeanplumInboxMessage
+        .createFromJsonMap("messageId##00", map);
+
+    assertEquals(false, message.isRead());
+
+    message.read();
+
+    assertEquals(true, message.isRead());
+  }
+
+  @Test
+  public void testImageURL() {
+    leanplumInbox.disableImagePrefetching();
+    ResponseHelper.seedResponse("/responses/newsfeed_response.json");
+    leanplumInbox.downloadMessages();
+    List<LeanplumInboxMessage> messagesList = leanplumInbox.allMessages();
+    LeanplumInboxMessage imageMessage = messagesList.get(0);
+
+    String actualUrl = imageMessage.getImageUrl().toString();
+
+    assertEquals("http://bit.ly/2GzJxxx",
+        actualUrl);
+  }
+
+  @Test
+  public void testImageFilepathIsReturnedIfPrefetchingEnabled() {
```

Comment:
It can be in the testImageURL test. You do not need to create a separate test for this. Just add a comment line.

Reply:
We should not combine these tests, since we want our unit tests to test a single code path. This test is testing a specific behavior that is different testImageURL.

Reply:
No they are testing to Different behaviors.

Reply:
"a single code path" - please check all others tests in Android SDK. Plus this is not good practice to setup all environment again if you can test it in previews one.

Reply:
1. I think we should probably look into all the other unit tests at some point, however I think for this sprint we should move forward and increase android automation coverage. 
2. I dont really understand your second sentence, could you please clarify?

Reply:
as I mention offline - please check AbstractTest.